server/repository/src/db_diesel/item.rs

```diff
@@ -227,6 +235,15 @@ fn create_filtered_query(store_id: String, filter: Option<ItemFilter>) -> BoxedI
             query = query.filter(item_dsl::is_vaccine.eq(is_vaccine));
         }
 
+        if let Some(category_id) = category_id {
+            let item_ids_for_category_id = item_category_join::table
+                .select(item_category_join::item_id)
+                .filter(item_category_join::category_id.eq(category_id.clone()))
+                .into_boxed();
+
+            query = query.filter(item_dsl::id.eq_any(item_ids_for_category_id));
```

Comment:
Was wondering what should happen with merged items?

I think that it's better to just leave it how it is though, as you don't want to find by the category of the non-merged item right?

Maybe add a comment to that effect?

Reply:
Yeah was wondering about this too, had a chat with Roxy... Seems like it's doing the right thing - if item was merged, we should find by the category of the "chosen" item - will add comment 🙏